### PR TITLE
Minor modernize updates

### PR DESCRIPTION
**Pull request by DavidSCN**
## Main changes of this PR
Related to #999. 

## Author's checklist

* <del> I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.</del>
* [x] I ran `tools/formatting/check-format` and everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.10.
* [x] squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?
* [ ] (more questions/tasks)

### CODE REVIEW

File: src/io/ExportVTKXML.cpp [hunk 1]
```diff
@@ -101,22 +101,22 @@ void ExportVTKXML::writeMasterFile(
 
   // write scalar data names
   outMasterFile << "      <PPointData Scalars=\"";
-  for (size_t i = 0; i < _scalarDataNames.size(); ++i) {
-    outMasterFile << _scalarDataNames[i] << ' ';
+  for (const auto &_scalarDataName : _scalarDataNames) {
+    outMasterFile << _scalarDataName << ' ';
```
Comment (oguzziya) [inline src/io/ExportVTKXML.cpp:105]: Maybe just `scalarDataName`, since it is not a member variable.

File: src/io/ExportVTKXML.cpp [hunk 2]
```diff
@@ -101,22 +101,22 @@ void ExportVTKXML::writeMasterFile(
...
   }
   // write vector data names
   outMasterFile << "\" Vectors=\"";
-  for (size_t i = 0; i < _vectorDataNames.size(); ++i) {
-    outMasterFile << _vectorDataNames[i] << ' ';
+  for (const auto &_vectorDataName : _vectorDataNames) {
+    outMasterFile << _vectorDataName << ' ';
```
Comment (oguzziya) [inline src/io/ExportVTKXML.cpp:110]: Same as above

File: src/io/ExportVTKXML.cpp [hunk 3]
```diff
@@ -101,22 +101,22 @@ void ExportVTKXML::writeMasterFile(
...
-  for (size_t i = 0; i < _scalarDataNames.size(); ++i) {
-    outMasterFile << "         <PDataArray type=\"Float64\" Name=\"" << _scalarDataNames[i] << "\" NumberOfComponents=\"" << 1 << "\"/>\n";
+  for (const auto &_scalarDataName : _scalarDataNames) {
+    outMasterFile << "         <PDataArray type=\"Float64\" Name=\"" << _scalarDataName << "\" NumberOfComponents=\"" << 1 << "\"/>\n";
   }
 
-  for (size_t i = 0; i < _vectorDataNames.size(); ++i) {
-    outMasterFile << "         <PDataArray type=\"Float64\" Name=\"" << _vectorDataNames[i] << "\" NumberOfComponents=\"" << 3 << "\"/>\n";
+  for (const auto &_vectorDataName : _vectorDataNames) {
+    outMasterFile << "         <PDataArray type=\"Float64\" Name=\"" << _vectorDataName << "\" NumberOfComponents=\"" << 3 << "\"/>\n";
```
Comment (oguzziya) [inline src/io/ExportVTKXML.cpp:119]: Same as above

File: src/io/ExportVTKXML.cpp [hunk 4]
```diff
@@ -240,12 +240,12 @@ void ExportVTKXML::exportData(
...
-  for (size_t i = 0; i < _scalarDataNames.size(); i++) {
-    outFile << _scalarDataNames[i] << ' ';
+  for (const auto &_scalarDataName : _scalarDataNames) {
+    outFile << _scalarDataName << ' ';
   }
   outFile << "\" Vectors=\"";
-  for (size_t i = 0; i < _vectorDataNames.size(); i++) {
-    outFile << _vectorDataNames[i] << ' ';
+  for (const auto &_vectorDataName : _vectorDataNames) {
+    outFile << _vectorDataName << ' ';
```
Comment (oguzziya) [inline src/io/ExportVTKXML.cpp:248]: Same as above

File: src/mapping/NearestProjectionMapping.cpp [hunk 5]
```diff
@@ -104,10 +104,10 @@ void NearestProjectionMapping::computeMapping()
 
   _interpolations.reserve(fVertices.size());
 
-  for (size_t i = 0; i < fVertices.size(); i++) {
+  for (const auto &fVertice : fVertices) {
```
Comment (oguzziya) [inline src/mapping/NearestProjectionMapping.cpp:107]: it should be `fVertex` i think

File: src/io/ExportVTKXML.cpp (hunk 1 above)
Comment (DavidSCN) [reply src/io/ExportVTKXML.cpp:105]: Ah right, `clang-tidy` has chosen the names automatically. I will fix it, good catch.

**Pull request merged**